pkgs/servers/http/ran/default.nix

```diff
@@ -0,0 +1,57 @@
+{ buildGoModule
+, fetchFromGitHub
+, lib
+, runCommand
+, ran
+, curl
+}:
+
+buildGoModule rec {
+  pname = "ran";
+  version = "0.1.6";
+  src = fetchFromGitHub {
+    owner = "m3ng9i";
+    repo = "ran";
+    rev = "v${version}";
+    hash = "sha256-iMvUvzr/jaTNdgHQFuoJNJnnkx2XHIUUlrPWyTlreEw=";
+  };
+
+  vendorSha256 = "sha256-ObroruWWNilHIclqNvbEaa7vwk+1zMzDKbjlVs7Fito=";
+
+  CGO_ENABLED = 0;
+
+  ldflags =
+    let
+      # Set a variety of useful meta variables for stamping the build with.
+      setVars = {
+        version = "v${version}";
+        branch = "master";
+      };
+      varFlags = with lib; concatStringsSep " "
+        (mapAttrsToList (name: value: "-X main._${name}_=${value}") setVars);
+    in
+    [
+      "${varFlags}"
+    ];
+  installPhase = ''
+    install -Dm0755 $GOPATH/bin/ran $out/bin/ran
+  '';
+
+  passthru.tests = {
+    simple = runCommand "ran-test" { } ''
+      echo hello world > index.html
+      ${ran}/bin/ran &
+      [ "$(${curl}/bin/curl 127.0.0.1:8080)" == "hello world" ]
+      kill %1
+      ${ran}/bin/ran --version > $out
+    '';
+  };
```

Comment:
still builds without it, not sure why they are here
```suggestion
```

Reply:
```suggestion
  CGO_ENABLED = 0;

  # Set a variety of useful meta variables for stamping the build with.
  ldflags = [
    "-X" "main._version_=v0.1.6"
    "-X" "main._branch_=master"
  ];
```

Reply:
should be fine without tests

Reply:
Yeah, good simplification of the ldflags. I pushed an update.
Why remove the passthru tests?

Reply:
tests failed on aarch64-linux, and failed once on x86_64-darwin, websockets are hard to test in a sandbox

https://logs.nix.ci/?key=nixos/nixpkgs.136439

Reply:
Is that a race condition on not starting up fast enough? Would a `sleep 1` or similar help? Or is this something specific to those architectures?

Context: I've been starting to try and include tests and make things "auto-update" friendly. If it just isn't worth it, that's fine, but I'd prefer to find a way to allow them.